pkg/controller/restore_controller.go

```diff
@@ -197,29 +197,40 @@ func (r *restoreReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	}
 
 	// add finalizer
-	if restore.DeletionTimestamp.IsZero() && !controllerutil.ContainsFinalizer(restore, ExternalResourcesFinalizer) {
-		original := restore.DeepCopy()
+	if original.DeletionTimestamp.IsZero() && !controllerutil.ContainsFinalizer(original, ExternalResourcesFinalizer) {
+		restore := original.DeepCopy()
 		controllerutil.AddFinalizer(restore, ExternalResourcesFinalizer)
 		if err := kubeutil.PatchResource(original, restore, r.kbClient); err != nil {
 			log.Errorf("fail to add finalizer: %s", err.Error())
 			return ctrl.Result{}, err
 		}
 	}
 
-	switch restore.Status.Phase {
+	switch original.Status.Phase {
 	case "", api.RestorePhaseNew:
 		// only process new restores
+	case api.RestorePhaseInProgress:
+		// if restore is in progress, we should not process it again
+		// we want to mark it as failed to avoid it being stuck in progress
+		// if so, mark it as failed, last loop did not successfully complete the restore
+		log.Debug("Restore has in progress status from prior reconcile, marking it as failed")
+		failedCopy := original.DeepCopy()
+		failedCopy.Status.Phase = api.RestorePhaseFailed
+		failedCopy.Status.FailureReason = "Restore from previous reconcile still in progress"
```

Comment:
We may want to suggest an APIServer failure here.
"Restore from previous reconcile still in progress. The API Server may have been down."